app/ui/browser-modern/views/browser/page/index.jsx

```diff
@@ -69,28 +69,35 @@ class Page extends Component {
       }));
     });
 
-    this.webview.addEventListener('did-navigate', () => {
+    this.webview.addEventListener('did-navigate', e => {
       // Event fired when a page loads, after `did-start-loading`.
       // Only emitted once.
-      const url = this.webview.getURL();
-      const isTofino = url.startsWith(Endpoints.TOFINO_PROTOCOL);
-      this.props.dispatch(PageActions.setPageDetails(this.props.pageId, { location: url }));
-      this.props.dispatch(UIEffects.setURLBarValue(this.props.pageId, url));
-      this.props.dispatch(UIEffects.focusURLBar(this.props.pageId, { select: isTofino }));
+      const pageId = this.props.pageId;
+      const isTofino = e.url.startsWith(Endpoints.TOFINO_PROTOCOL);
+      const history = this.webview.getWebContents().history;
+      const historyIndex = this.webview.getWebContents().getActiveIndex();
```

Comment:
I've done no measurements so can't say but the main process is generally not doing anything so I expect it is unlikely to cause significant jank. It's worth noting that many of the methods on webview cause sync IPC calls to the main process so generally we shouldn't expect it to be cheap to interact with.